File: Algorithms/Sorting/Counting Sort/Counting Sort.cpp
void counting_sort(int arr[], int min, int max, int n)
{
    int range = max - min + 1, acc = 0;
    int count[range] = {0};

    for (int i = 0; i < n; i++)
    {
        int index = arr[i] - min;
        count[index]++;
    }

        for (int i = 0; i < range; i++)
        for (int j = 0; j < count[i]; j++)
        {
            arr[acc++] = min + i;
        }
}

File: Algorithms/Sorting/Counting Sort/Counting Sort.py
def counting_sort(arr, min, max, n):
    count = [0 for _ in range(max - min + 1)]

    for i in range(n):
        index = arr[i] - min
        count[index] += 1

    acc = 0
    for i, c in enumerate(count):
        for _ in range(c):
            arr[acc] = min + i
            acc += 1
